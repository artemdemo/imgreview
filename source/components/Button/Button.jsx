/* eslint-disable no-trailing-spaces */
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import './Button.less';

const Button = (props) => {
    const buttonClass = classnames({
        button: true,
        button_primary: props.primary,
        button_secondary: props.secondary,
        button_disabled: props.disabled,
    });
    return (
        <button
            title={props.title}
            onClick={props.onClick}
            className={buttonClass}
            disabled={props.disabled}
            type={props.type}
        >
            {props.children}
        </button>
    );
};

Button.propTypes = {
    title: PropTypes.string,
    type: PropTypes.string,
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    disabled: PropTypes.bool,
    onClick: PropTypes.func,
};

Button.defaultProps = {
    title: null,
    type: 'button',
    primary: true,
    secondary: false,
    disabled: false,
    onClick: null,
};

export default Button;
