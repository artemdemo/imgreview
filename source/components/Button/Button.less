@import '../../styles/functions';

.button {
    .resetButton();
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    user-select: none;
    border: 1px solid transparent;
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: 0.25rem;
}
    .button_primary {
        color: #fff;
        background-color: #007bff;
        border-color: #007bff;
    }
    .button_secondary {
        color: #fff;
        background-color: #6c757d;
        border-color: #6c757d;
    }
