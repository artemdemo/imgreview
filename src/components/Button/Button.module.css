.Button {
  display: inline-block;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
}
.Button_block {
  width: 100%;
}
.Button_primary {
  color: #fff;
  background-color: #007bff;
  border-color: #007bff;
}
.Button_secondary {
  color: #fff;
  background-color: #6c757d;
  border-color: #6c757d;
}
.Button_disabled {
  color: rgba(255, 255, 255, 0.5);
  background-color: rgba(108, 117, 125, 0.6);
  border-color: transparent;
  cursor: not-allowed;
}
