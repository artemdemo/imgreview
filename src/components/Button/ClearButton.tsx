import React from 'react';
import styled, { css } from 'styled-components';

export const clearButtonCss = css`
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;

  background: transparent;

  cursor: pointer;

  /* inherit font & color from ancestor */
  color: inherit;
  font: inherit;

  /* Normalize 'line-height'. Cannot be changed from 'normal' in Firefox 4+. */
  line-height: normal;

  /* Corrects font smoothing for webkit */
  -webkit-font-smoothing: inherit;
  -moz-osx-font-smoothing: inherit;

  /* Corrects inability to style clickable 'input' types in iOS */
  -webkit-appearance: none;
`;

const ClearButton = styled.button`
  ${clearButtonCss}
`;

export default ClearButton;
