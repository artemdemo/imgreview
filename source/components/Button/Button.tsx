/* eslint-disable no-trailing-spaces */
import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components'
import ClearButton from './ClearButton';

const Button = styled(ClearButton)`
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    user-select: none;
    border: 1px solid transparent;
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: 0.25rem;
    ${props => props.primary && `
        color: #fff;
        background-color: #007bff;
        border-color: #007bff;
    `}
    ${props => props.secondary && `
        color: #fff;
        background-color: #6c757d;
        border-color: #6c757d;
    `}
    ${props => props.disabled && `
        color: rgba(255, 255, 255, 0.5);
        background-color: rgba(108, 117, 125, 0.6);
        border-color: transparent;
        cursor: not-allowed;
    `}
`;

Button.propTypes = {
    type: PropTypes.string,
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    disabled: PropTypes.bool,
};

Button.defaultProps = {
    type: 'button',
    primary: true,
    secondary: false,
    disabled: false,
};

export default Button;
