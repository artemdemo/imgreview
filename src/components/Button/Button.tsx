/* eslint-disable no-trailing-spaces */
import React from 'react';
import styled from 'styled-components'
import ClearButton from './ClearButton';

export enum EButtonAppearance {
    PRIMARY,
    SECONDARY,
}

type TProps = {
    appearance?: EButtonAppearance;
    disabled: boolean;
};

const Button = styled(ClearButton)<TProps>`
    display: inline-block;
    font-weight: 400;
    text-align: center;
    white-space: nowrap;
    user-select: none;
    border: 1px solid transparent;
    padding: 0.375rem 0.75rem;
    font-size: 1rem;
    line-height: 1.5;
    border-radius: 0.25rem;
    ${props => (!props.appearance || props.appearance === EButtonAppearance.PRIMARY) && `
        color: #fff;
        background-color: #007bff;
        border-color: #007bff;
    `}
    ${props => props.appearance === EButtonAppearance.SECONDARY && `
        color: #fff;
        background-color: #6c757d;
        border-color: #6c757d;
    `}
    ${props => props.disabled && `
        color: rgba(255, 255, 255, 0.5);
        background-color: rgba(108, 117, 125, 0.6);
        border-color: transparent;
        cursor: not-allowed;
    `}
`;

Button.defaultProps = {
    type: 'button',
    disabled: false,
};

export default Button;
