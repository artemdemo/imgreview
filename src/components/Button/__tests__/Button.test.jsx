import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Button from '../Button.tsx';

describe('Button', () => {
    it('empty render', () => {
        const tree = renderer.create(
            <Button />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with text', () => {
        const tree = renderer.create(
            <Button>Some text</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with title', () => {
        const tree = renderer.create(
            <Button title='some title'>Some text</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with primary property', () => {
        const tree = renderer.create(
            <Button primary>Some text</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with secondary property', () => {
        const tree = renderer.create(
            <Button secondary>Some text</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render disabled', () => {
        const tree = renderer.create(
            <Button disabled>Some text</Button>
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should handle onClick', () => {
        const onClickMock = jest.fn();
        const wrapper = mount(
            <Button onClick={onClickMock}>Some text</Button>,
        );
        wrapper.find('button').simulate('click');
        expect(onClickMock).toBeCalled();
    });
});
