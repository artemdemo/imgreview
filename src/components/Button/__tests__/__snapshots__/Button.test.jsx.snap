// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button empty render 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH ihcTYM"
  disabled={false}
  type="button"
/>
`;

exports[`Button should render disabled 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH fTYagl"
  disabled={true}
  type="button"
>
  Some text
</button>
`;

exports[`Button should render with primary property 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH ihcTYM"
  disabled={false}
  type="button"
>
  Some text
</button>
`;

exports[`Button should render with secondary property 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH kJBRUV"
  disabled={false}
  type="button"
>
  Some text
</button>
`;

exports[`Button should render with text 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH ihcTYM"
  disabled={false}
  type="button"
>
  Some text
</button>
`;

exports[`Button should render with title 1`] = `
<button
  className="sc-bdVaJa sc-bwzfXH ihcTYM"
  disabled={false}
  title="some title"
  type="button"
>
  Some text
</button>
`;
